internal/controller/api/management.go

```diff
@@ -125,10 +130,66 @@ func (s *ManagementServer) handleConnectionStatus() http.HandlerFunc {
 			connectionStatus.Status = DISCONNECTED_STATUS
 		}
 
-		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
-		w.WriteHeader(http.StatusOK)
-		if err := json.NewEncoder(w).Encode(connectionStatus); err != nil {
+		WriteJSONResponse(w, http.StatusOK, connectionStatus)
+	}
+}
+
+func (s *ManagementServer) handleConnectionPing() http.HandlerFunc {
+
+	return func(w http.ResponseWriter, req *http.Request) {
+
+		var connID connectionID
+
+		body, err := ioutil.ReadAll(io.LimitReader(req.Body, 1048576))
+
+		if err != nil {
 			panic(err)
 		}
+
+		if err := req.Body.Close(); err != nil {
+			panic(err)
+		}
+
+		if err := json.Unmarshal(body, &connID); err != nil {
+			errorResponse := errorResponse{Title: "Unable to process json input",
+				Status: http.StatusUnprocessableEntity,
+				Detail: err.Error()}
+			WriteJSONResponse(w, errorResponse.Status, errorResponse)
+			return
+		}
+
+		log.Println(connID)
+
+		var payload interface{}
+
+		client := s.connectionMgr.GetConnection(connID.Account, connID.NodeID)
+		if client == nil {
+			errorResponse := errorResponse{Title: "No connection found to node",
+				Status: http.StatusBadRequest,
+				Detail: "No connection found to node"}
+			WriteJSONResponse(w, errorResponse.Status, errorResponse)
+			return
+		}
+
+		payload, err = client.Ping(req.Context(), connID.NodeID, []string{connID.NodeID})
```

Comment:
Just a note from the call so I don't forget. Probably need to generate the routes instead of requiring them to be passed in.

Reply:
Agreed.